src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.LowLevelLifoSemaphore.Unix.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Threading
+{
+    internal static partial class ClrThreadPool
+    {
+        /// <summary>
+        /// A LIFO semaphore (used by the thread pool to keep hot threads hot and cold threads cold).
```

Comment:
It may be good to mention that like other "low level" sync objects, waits on this object are not interruptible